introspection-engine/introspection-engine-tests/tests/re_introspection/relation_mode/mssql.rs

```diff
@@ -0,0 +1,361 @@
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+          provider     = "sqlserver"
+          url          = env("TEST_DATABASE_URL")
+          relationMode = "prisma"
+        }
+
+        model Foo {
+          id     Int @id
+          bar    Bar @relation(fields: [bar_id], references: [id])
+          bar_id Int @unique
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
+
+    let result = api.re_introspect_config(input).await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+// relationMode="foreignKeys" preserves the relation policy ("foreignKeys") as well as @relations, which are moved to the bottom.
+#[test_connector(tags(Mssql))]
+async fn relation_mode_foreign_keys(api: &TestApi) -> TestResult {
+    let init = formatdoc! {r#"
+        CREATE TABLE [dbo].[Foo] (
+            [id] INT NOT NULL,
+            [bar_id] INT NOT NULL,
+            CONSTRAINT [Foo_pkey] PRIMARY KEY CLUSTERED ([id]),
+            CONSTRAINT [Foo_bar_id_key] UNIQUE NONCLUSTERED ([bar_id])
+        );
+        
+        CREATE TABLE [dbo].[Bar] (
+            [id] INT NOT NULL,
+            CONSTRAINT [Bar_pkey] PRIMARY KEY CLUSTERED ([id])
+        );
+        
+        ALTER TABLE [dbo].[Foo] ADD CONSTRAINT [Foo_bar_id_fkey] FOREIGN KEY ([bar_id]) REFERENCES [dbo].[Bar]([id]) ON DELETE NO ACTION ON UPDATE CASCADE;
+    "#};
+
+    api.raw_cmd(&init).await;
+
+    let input = indoc! {r#"
+        generator client {
+            provider        = "prisma-client-js"
+            previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+            provider     = "sqlserver"
+            url          = env("TEST_DATABASE_URL")
+            relationMode = "foreignKeys"
+        }
+
+        model Foo {
+            id     Int @id
+            bar    Bar @relation(fields: [bar_id], references: [id])
+            bar_id Int @unique
+        }
+
+        model Bar {
+            id  Int  @id
+            foo Foo?
+        }
+    "#};
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["referentialIntegrity"]
+        }
+
+        datasource db {
+          provider     = "sqlserver"
+          url          = env("TEST_DATABASE_URL")
+          relationMode = "foreignKeys"
+        }
+
+        model Foo {
+          id     Int @id
+          bar_id Int @unique
+          bar    Bar @relation(fields: [bar_id], references: [id])
+        }
+
+        model Bar {
+          id  Int  @id
+          foo Foo?
+        }
+    "#]];
+
+    let result = api.re_introspect_config(input).await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+// @relations are moved to the bottom of the model even when no referentialIntegrity/relationMode is used.
+#[test_connector(tags(Mssql))]
+async fn no_relation_mode(api: &TestApi) -> TestResult {
```

Comment:
This test does not test something relation mode specific — it's useful in the process of figuring out if that behaviour was influenced by relationMode (or not, in this case) but out of context I don't know if we will be able to make sense of it next time we visit this file.

Reply:
Fair enough, it was indeed useful initially, but now I agree we could drop it safely.